Linker errors on macOS

The underlying `swift-actions/setup-swift` action is fragile on macOS and causes linker errors when using the system Swift libraries:
```

1.	Apple Swift version 5.9 (swift-5.9-RELEASE)
2.	Compiling with the current language version
3.	While evaluating request IRGenRequest(IR Generation for file "/Users/runner/work/SwiftGLib/SwiftGLib/SwiftGLib/.build/plugins/outputs/swiftglib/GLib/gir2swift-plugin/gir2swift-generated/GLib/GLib-2.0-functions.swift")
4.	While emitting IR SIL function "@$s4GLib19buildFilenameValist12firstElement4argsSSSgSPys4Int8VGSg_SpySo13__va_list_tagVGSgtF".
 for 'buildFilenameValist(firstElement:args:)' (at /Users/runner/work/SwiftGLib/SwiftGLib/SwiftGLib/.build/plugins/outputs/swiftglib/GLib/gir2swift-plugin/gir2swift-generated/GLib/GLib-2.0-functions.swift:1323:19)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000107a8f6c7 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 39
1  swift-frontend           0x0000000107a8eaa5 llvm::sys::RunSignalHandlers() + 85
2  swift-frontend           0x0000000107a8fd10 SignalHandler(int) + 288
3  libsystem_platform.dylib 0x00007ff8158565ed _sigtramp + 29
4  libsystem_platform.dylib 0x0000000000000014 _sigtramp + 18446603370219870788
5  libsystem_c.dylib        0x00007ff81574fb45 abort + 123
6  swift-frontend           0x0000000107b60c38 (anonymous namespace)::IRGenDebugInfoImpl::getOrCreateType(swift::irgen::DebugTypeInfo) (.cold.8) + 200
7  swift-frontend           0x000000010277592f (anonymous namespace)::IRGenDebugInfoImpl::getOrCreateType(swift::irgen::DebugTypeInfo) + 5551
8  swift-frontend           0x000000010277608b (anonymous namespace)::IRGenDebugInfoImpl::createParameterType(llvm::SmallVectorImpl<llvm::Metadata*>&, swift::SILType) + 107
9  swift-frontend           0x0000000102775d7d (anonymous namespace)::IRGenDebugInfoImpl::createParameterTypes(swift::CanTypeWrapper<swift::SILFunctionType>) + 541
10 swift-frontend           0x000000010276df1e (anonymous namespace)::IRGenDebugInfoImpl::emitFunction(swift::SILDebugScope const*, llvm::Function*, swift::SILFunctionTypeRepresentation, swift::SILType, swift::DeclContext*, llvm::StringRef) + 2606
11 swift-frontend           0x000000010276e8cc (anonymous namespace)::IRGenDebugInfoImpl::emitFunction(swift::SILFunction&, llvm::Function*) + 188
12 swift-frontend           0x00000001027b7669 (anonymous namespace)::IRGenSILFunction::emitSILFunction() + 969
13 swift-frontend           0x00000001027b6c33 swift::irgen::IRGenModule::emitSILFunction(swift::SILFunction*) + 1939
14 swift-frontend           0x0000000102637325 swift::irgen::IRGenerator::emitGlobalTopLevel(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) + 869
15 swift-frontend           0x00000001027570ba swift::IRGenRequest::evaluate(swift::Evaluator&, swift::IRGenDescriptor) const + 1818
16 swift-frontend           0x00000001027b5f5d swift::GeneratedModule swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::callDerived<0ul>(swift::Evaluator&, std::__1::integer_sequence<unsigned long, 0ul>) const + 221
17 swift-frontend           0x00000001027b5e3e swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::IRGenRequest const&, swift::Evaluator&) + 14
18 swift-frontend           0x0000000102766cd4 llvm::Expected<swift::IRGenRequest::OutputType> swift::Evaluator::getResultUncached<swift::IRGenRequest>(swift::IRGenRequest const&) + 388
19 swift-frontend           0x0000000102759b8f swift::performIRGeneration(swift::FileUnit*, swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::StringRef, swift::PrimarySpecificPaths const&, llvm::StringRef, llvm::GlobalVariable**) + 319
20 swift-frontend           0x00000001022b59ae performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 1710
21 swift-frontend           0x00000001022b4acb swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 827
22 swift-frontend           0x00000001022b6f85 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 4085
23 swift-frontend           0x000000010210f164 swift::mainEntry(int, char const**) + 1268
24 dyld                     0x00007ff8154cf41f start + 1903
Error: Process completed with exit code 1.
```

Better to use [ maxim-lobanov/setup-xcode@v1](https://github.com/maxim-lobanov/setup-xcode) on macOS.

--- a/README.md
+++ b/README.md
@@ -1,5 +1,124 @@
-install-guunits
-============
-*A github action for installing guunits onto the system*
+Create GitHub Pages Documentation using DocC
+============================================
 
----
+<p>
+  <a href="https://github.com/features/actions">
+    <img src="https://img.shields.io/badge/GitHub-Action-blue?logo=github" alt="GitHub Action" />
+  </a>
+  <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources">
+    <img src="https://img.shields.io/badge/platform-macOS%20%7C%20Linux-darkgreen" alt="Supports macOS and Linux" />
+  </a>
+  <a href="https://swift.org">
+    <img src="https://img.shields.io/badge/Swift-5.9-F05138?logo=swift&logoColor=white" alt="Swift 5.9" />
+  </a>
+  <a href="https://github.com/mipalgu/swiftpm-generate-documentation/releases/latest">
+    <img src="https://img.shields.io/github/v/release/mipalgu/swiftpm-generate-documentation?sort=semver" alt="Latest release" />
+  </a>
+</p>
+
+*A [GitHub Action](https://github.com/features/actions) for generating DocC documentation and deploying to GitHub Pages*
+
+## Usage
+
+To run the action to generate documentation with the latest Xcode or Swift version available
+and deploy to GitHub pages, add the action as a step in your workflow:
+
+```yaml
+- uses: mipalgu/swiftpm-generate-documentation@main
+```
+
+A specific Xcode or Swift version can be set using the `swift-version` input:
+
+```yaml
+- uses: mipalgu/swiftpm-generate-documentation@main
+  with:
+    swift-version: "5.9"
+```
+
+The following section shows the input parameter variables that can be configured.
+
+## Configuration
+
+Here are the input parameters (and their default values) that can be set up in the YAML configuration:
+
+```yaml
+inputs:
+  output-path:
+    description: The path containing the generated docs.
+    required: false
+    default: "./docs"
+  working-directory:
+    description: The directory containing the swift package.
+    required: false
+    default: ""
+  hosting-base-path:
+    description: The the website will be hosted.
+    required: false
+    default: ${{ github.event.repository.name }}
+  minimum-access-level:
+    description: The minimum access level that must be included in the docs.
+    default: "public"
+  swift-version:
+    description: The version of Xcode/Swift used to execute the docC package plugin.
+    default: latest
+```
+
+## Note about versions
+
+This project uses [swift-actions/setup-swift@v1](https://github.com/swift-actions/setup-swift)
+to set up Swift on Linux and
+[maxim-lobanov/setup-xcode@v1](https://github.com/maxim-lobanov/setup-xcode)
+to set up Xcode on macOS.
+See the corresponding actions for details on their versioning system, but it is important to note
+that they use strict semantic versioning to determine what version of Swift to configure.
+In particular, for [setup-swift](https://github.com/swift-actions/setup-swift)
+this differs from the official convention used by Swift, in that if a patch version
+is omitted, the lates patch version is used.
+
+For example, there is a Swift version `5.9` but using this as value for `swift-version`
+will be interpreted as a version _range_ of `5.9.x` where `x` will match the latest patch version
+available for that major and minor version.
+
+In other words specifying...
+- `"5.9.0"` will resolve to version `5.9`
+- `"5.9"` will resolve to latest patch version (e.g. `5.9.2`)
+- `"5"` will resolve to latest minor and patch version (e.g. `5.9.x`)
+
+### Caveats
+
+Quote your inputs as YAML interprets eg. `5.0` as a float, causing GitHub actions to then interpret that as `5`,
+which will result in eg. Swift 5.9.2 being resolved. Thus:
+
+```
+- uses: mipalgu/swiftpm-generate-documentation@main
+  with:
+    swift-version: '5.0'
+```
+
+Not:
+
+```
+- uses: mipalgu/swiftpm-generate-documentation@main
+  with:
+    swift-version: 5.0
+```
+
+## Keeping the action up-to-date
+
+You have two options for keeping this action up-to-date: either you use the main branch (i.e. `@main`)
+or you define a specific version (such as `@v1.2.3`) or use the major version tag (as in `@v1`).
+
+### Dependabot
+
+We recommend using a specific version tag together with
+[Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates) to keep the action up-to-date. This way, you will automatically get notified when the action updates
+and you can read the ChangeLog directly in the PR opened by dependabot.
+
+### Main Version Tag
+
+If you don't plan on keeping tabs on updates or don't want to use Dependabot but still would like to always use the latest version, you can use the `@main` version tag.
+
+## Legal
+See [LICENSE.txt](https://raw.githubusercontent.com/rhx/swiftpm-generate-documentation/main/LICENSE.txt) for details. 
+The Swift logo is a trademark of Apple Inc.
+

--- a/action.yml
+++ b/action.yml
@@ -18,14 +18,22 @@ inputs:
     description: The minimum access level that must be included in the docs.
     default: "public"
   swift-version:
-    description: The version of swift used to execute the docC package plugin.
-    required: true
+    description: The version of Xcode/Swift used to execute the docC package plugin.
+    default: latest
 
 runs:
   using: 'composite'
 
   steps:
-      - uses: swift-actions/setup-swift@v1
+      - name: Setup Xcode
+        if: ${{ runner.os == 'macOS' }}
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: ${{ inputs.swift-version }}
+
+      - name: Setup Swift
+        if: ${{ runner.os == 'Linux' }}
+        uses: actions/setup-swift@v1
         with:
           swift-version: ${{ inputs.swift-version }}
 

--- a/action.yml
+++ b/action.yml
@@ -33,7 +33,7 @@ runs:
 
       - name: Setup Swift
         if: ${{ runner.os == 'Linux' }}
-        uses: actions/setup-swift@v1
+        uses: swift-actions/setup-swift@v1
         with:
           swift-version: ${{ inputs.swift-version }}
 

--- a/action.yml
+++ b/action.yml
@@ -3,7 +3,7 @@ description: Generates a documentation website from a swift package.
 
 inputs:
   output-path:
-    decription: The path containing the generated docs.
+    description: The path containing the generated docs.
     required: false
     default: "./docs"
   working-directory:

--- a/src/main.swift
+++ b/src/main.swift
@@ -197,7 +197,7 @@ do {
         let packageDirectory = baseDirectory + "/package"
         let packageIndex = packageDirectory + "/index.html"
         let redirectionPath: String
-        if fm.fileExists(atPath: packageIndex) {
+        if fm.fileExists(atPath: outputPath + packageIndex) {
             redirectionPath = packageIndex
         } else {
             redirectionPath = baseDirectory

--- a/src/main.swift
+++ b/src/main.swift
@@ -162,7 +162,7 @@ func parsePath(_ argument: String, relativeTo basePath: URL) -> URL? {
 }
 
 #if os(macOS)
-let swiftBinDefault = "/Users/runner/hostedtoolcache/swift-macOS/5.6.3/x64/usr/bin/swift"
+let swiftBinDefault = "/usr/bin/swift"
 #else
 let swiftBinDefault = "/opt/hostedtoolcache/swift-Ubuntu/5.6.3/x64/usr/bin/swift"
 #endif

--- a/src/main.swift
+++ b/src/main.swift
@@ -192,17 +192,18 @@ do {
     try package.generateDocumentation(swiftBin: swiftBin, hostingBasePath: hostingBasePath, outputPath: outputURL, minimumAccessLevel: minimumAccessLevel)
     let indexURL = outputURL.appendingPathComponent("index.html", isDirectory: false)
     if package.targets.filter({ $0.type != .unsupported }).count == 1, let first = package.targets.first(where: { $0.type != .unsupported }) {
-        let documentationDirectory = "/" + (hostingBasePath.map { $0 + "/" } ?? "") + first.name + "/documentation"
+        let hostingDirectory = "/" + (hostingBasePath.map { $0 + "/" } ?? "")
+        let documentationDirectory = first.name + "/documentation"
         let baseDirectory = documentationDirectory + "/" + first.name.lowercased()
         let packageDirectory = baseDirectory + "/package"
         let packageIndex = packageDirectory + "/index.html"
-        let packageIndexPath = outputURL.path + packageIndex
+        let packageIndexPath = outputURL.path + "/" + packageIndex
         let redirectionPath: String
         if fm.fileExists(atPath: packageIndexPath) {
-            redirectionPath = packageIndex
+            redirectionPath = hostingDirectory + packageIndex
             print("Redirecting to '\(redirectionPath)'.")
         } else {
-            redirectionPath = baseDirectory
+            redirectionPath = hostingDirectory + baseDirectory
             print("\(packageIndexPath) not found, redirecting to '\(redirectionPath)'.")
         }
         let content = """

--- a/src/main.swift
+++ b/src/main.swift
@@ -172,7 +172,7 @@ let currentDirectory = URL(fileURLWithPath: FileManager.default.currentDirectory
 let minimumAccessLevel = parseArg("--minimum-access-level") ?? "public"
 let hostingBasePath = parseArg("--hosting-base-path").flatMap { $0 == "/" ? nil : $0 }
 let workingDirectory = parsePath("--working-directory", relativeTo: currentDirectory) ?? currentDirectory
-let outputPath = parsePath("--output-path", relativeTo: workingDirectory) ?? workingDirectory.appendingPathComponent("docs", isDirectory: true)
+let outputURL = parsePath("--output-path", relativeTo: workingDirectory) ?? workingDirectory.appendingPathComponent("docs", isDirectory: true)
 
 let fm = FileManager.default
 fm.changeCurrentDirectoryPath(workingDirectory.path)
@@ -188,16 +188,16 @@ do {
     fatalError("Error parsing Package.swift: " + e.localizedDescription)
 }
 do {
-    try fm.createDirectory(at: outputPath, withIntermediateDirectories: true)
-    try package.generateDocumentation(swiftBin: swiftBin, hostingBasePath: hostingBasePath, outputPath: outputPath, minimumAccessLevel: minimumAccessLevel)
-    let indexURL = outputPath.appendingPathComponent("index.html", isDirectory: false)
+    try fm.createDirectory(at: outputURL, withIntermediateDirectories: true)
+    try package.generateDocumentation(swiftBin: swiftBin, hostingBasePath: hostingBasePath, outputPath: outputURL, minimumAccessLevel: minimumAccessLevel)
+    let indexURL = outputURL.appendingPathComponent("index.html", isDirectory: false)
     if package.targets.filter({ $0.type != .unsupported }).count == 1, let first = package.targets.first(where: { $0.type != .unsupported }) {
         let documentationDirectory = "/" + (hostingBasePath.map { $0 + "/" } ?? "") + first.name + "/documentation"
         let baseDirectory = documentationDirectory + "/" + first.name.lowercased()
         let packageDirectory = baseDirectory + "/package"
         let packageIndex = packageDirectory + "/index.html"
         let redirectionPath: String
-        if fm.fileExists(atPath: outputPath + packageIndex) {
+        if fm.fileExists(atPath: outputURL.path + packageIndex) {
             redirectionPath = packageIndex
         } else {
             redirectionPath = baseDirectory

--- a/src/main.swift
+++ b/src/main.swift
@@ -200,7 +200,7 @@ do {
         let packageIndexPath = outputURL.path + "/" + packageIndex
         let redirectionPath: String
         if fm.fileExists(atPath: packageIndexPath) {
-            redirectionPath = hostingDirectory + packageIndex
+            redirectionPath = hostingDirectory + packageDirectory
             print("Redirecting to '\(redirectionPath)'.")
         } else {
             redirectionPath = hostingDirectory + baseDirectory

--- a/src/main.swift
+++ b/src/main.swift
@@ -188,17 +188,26 @@ do {
     fatalError("Error parsing Package.swift: " + e.localizedDescription)
 }
 do {
-    try FileManager.default.createDirectory(at: outputPath, withIntermediateDirectories: true)
+    try fm.createDirectory(at: outputPath, withIntermediateDirectories: true)
     try package.generateDocumentation(swiftBin: swiftBin, hostingBasePath: hostingBasePath, outputPath: outputPath, minimumAccessLevel: minimumAccessLevel)
     let indexURL = outputPath.appendingPathComponent("index.html", isDirectory: false)
     if package.targets.filter({ $0.type != .unsupported }).count == 1, let first = package.targets.first(where: { $0.type != .unsupported }) {
         let documentationDirectory = "/" + (hostingBasePath.map { $0 + "/" } ?? "") + first.name + "/documentation"
+        let baseDirectory = documentationDirectory + "/" + first.name.lowercased()
+        let packageDirectory = baseDirectory + "/package"
+        let packageIndex = packageDirectory + "/index.html"
+        let redirectionPath: String
+        if fm.fileExists(atPath: packageIndex) {
+            redirectionPath = packageIndex
+        } else {
+            redirectionPath = baseDirectory
+        }
         let content = """
             <!DOCTYPE html>
             <html lang="en">
             <head>
                 <title>\(first.name)</title>
-                <meta http-equiv = "refresh" content = "0; url = \(documentationDirectory)/\(first.name.lowercased())" />
+                <meta http-equiv = "refresh" content = "0; url = \(redirectionPath)" />
             </head>
             <body>
                 <p>Redirecting</p>

--- a/src/main.swift
+++ b/src/main.swift
@@ -196,11 +196,14 @@ do {
         let baseDirectory = documentationDirectory + "/" + first.name.lowercased()
         let packageDirectory = baseDirectory + "/package"
         let packageIndex = packageDirectory + "/index.html"
+        let packageIndexPath = outputURL.path + packageIndex
         let redirectionPath: String
-        if fm.fileExists(atPath: outputURL.path + packageIndex) {
+        if fm.fileExists(atPath: packageIndexPath) {
             redirectionPath = packageIndex
+            print("Redirecting to '\(redirectionPath)'.")
         } else {
             redirectionPath = baseDirectory
+            print("\(packageIndexPath) not found, redirecting to '\(redirectionPath)'.")
         }
         let content = """
             <!DOCTYPE html>

